### PR TITLE
Handle remove collateral if account has expired short (#109)

### PR DESCRIPTION
**Pull request by kjr217**
Responding to Issue #109. Solution here reverts the remove collateral if there is an expired short position, a better solution could be to settle the expired option automatically if an attempt to remove collateral was made as presumably the user would rather do that. I think the second solution is worth doing @antoncoding

### CODE REVIEW

**Review by antoncoding — APPROVED** [review]

File: src/core/engines/advanced-margin/AdvancedMarginEngine.sol [hunk 1]
```diff
@@ -332,7 +332,18 @@
     ) internal {
         // decode parameters
         (uint80 amount, address recipient, uint8 collateralId) = abi.decode(_data, (uint80, address, uint8));
-
+        // check if there is an expired short still in the account, if there is then collateral cant be removed
+        // until the position is settled
+        if (_account.shortCallAmount > 0) {
+            (,, uint64 expiry,,) = TokenIdUtil.parseTokenId(_account.shortCallId);
```
[the remark this answers is not in the document]
Comment (antoncoding) [reply src/core/engines/advanced-margin/AdvancedMarginEngine.sol:338]: I believe this is false positive

**Comment by antoncoding** [comment]
LGTM, just need to run `npm lint` before merging

**Comment by kjr217** [comment]
Do you think we should explore auto-settling instead?

**Comment by kjr217** [comment]
Ill also write tests for these too before merging

**Comment by antoncoding** [comment]
> Do you think we should explore auto-settling instead?

yeah that would be a good idea

The easiest implementation I can think of is to wrap some logic in `_settle` into another internal function and just call it if shortAmount > 0.

**Comment by antoncoding** [comment]
Actually can you wait a bit on this? #114 is gonna change the function structure quite a lot and is going to affect this PR, we plan to merge that one today. 
So you can wait until that one is merged to avoid some conflicts.

If you want I can help you rebase this branch to master first after merging.

**Comment by antoncoding** [comment]
rebased at #174

**Pull request closed without merging**